parla/task_runtime.py

```diff
@@ -973,45 +958,76 @@ def shuffled(lst: Iterable[_T]) -> List[_T]:
     return lst
 
 class Scheduler(ControllableThread, SchedulerContext):
+    # See __init__ function below for comments on the functionality of these members
     _environments: TaskEnvironmentRegistry
-    # Worker threads for each device.
     _worker_threads: List[WorkerThread]
+    _free_worker_threads: Deque[WorkerThread]
     _available_resources: ResourcePool
-    period: float
-    max_worker_queue_depth: int
+    period: float # TODO: Figure out what this is for
```

Comment:
now we don't need this. This was for re-enqueueing delay. If unassigned allocation was not enough, scheduler re-enqueue the task to the queue after this period time.

Reply:
I think we may still end up using `period` to set a dynamically adjustable period on the scheduler loop though? I guess that's TBD. No harm removing it for now.